lib/src/dependency.dart

```diff
@@ -107,12 +110,12 @@ class SdkDependency extends Dependency {
 class GitDependency extends Dependency {
   @JsonKey(fromJson: parseGitUri, required: true, disallowNullValue: true)
   final Uri url;
-  final String ref;
-  final String path;
+  final String? ref;
+  final String? path;
 
   GitDependency(this.url, this.ref, this.path) : super._();
```

Comment:
a bit weird to have nullable fields be non-optional args 🤷‍♂️

Reply:
ack